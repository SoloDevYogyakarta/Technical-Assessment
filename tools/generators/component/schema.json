{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "component",
  "type": "object",
  "properties": {
    "filename": {
      "type": "string",
      "description": "Write your file name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your filename"
      },
      "$default": {
        "$source": "filename"
      }
    },
    "directoryname": {
      "type": "string",
      "description": "Write your directory name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your directoryname"
      },
      "$default": {
        "$source": "directoryname"
      }
    },
    "projectname": {
      "type": "string",
      "description": "Write your project name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your projectname"
      },
      "$default": {
        "$source": "projectname"
      }
    }
  }
}
