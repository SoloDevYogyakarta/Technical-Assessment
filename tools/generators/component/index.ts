import {
  Tree,
  formatFiles,
  getProjects,
  names,
  generateFiles,
} from '@nrwl/devkit';
import { join } from 'path';
import { ISchema } from '../schema';

export default async function (tree: Tree, schema: ISchema) {
  const workspace = getProjects(tree).get(schema.projectname);
  const templatePath = 'tools/generators/component/templates';
  const directoryPath = schema.directoryname
    ? `${workspace?.sourceRoot}/lib/${schema.directoryname}`
    : `${workspace?.sourceRoot}/lib`;
  const filename = `${schema.filename}.component`;
  const targetPath = join(directoryPath, filename);
  const componentNames = names(filename);
  const substitutions = {
    tmpl: '',
    ...componentNames,
  };
  generateFiles(tree, templatePath, targetPath, substitutions);
  await formatFiles(tree);
}
