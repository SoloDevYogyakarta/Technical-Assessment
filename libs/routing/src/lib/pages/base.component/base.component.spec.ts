import { BaseComponent } from './base.component';
import {
  TestBed,
  TestBedStatic,
  ComponentFixture,
} from '@angular/core/testing';

describe('BaseComponent', () => {
  let container: TestBedStatic;
  let component: ComponentFixture<BaseComponent>;

  beforeEach(async () => {
    container = await TestBed.configureTestingModule({
      imports: [],
      declarations: [BaseComponent],
    }).compileComponents();

    component = TestBed.createComponent(BaseComponent);
    component.detectChanges();
  });

  it('should render snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it("should have as title 'BaseComponent'", () => {
    const template = component.componentInstance;
    expect(template.title).toEqual('BaseComponent');
  });

  it('should render title', () => {
    const template = component.nativeElement as HTMLElement;
    expect(template.querySelector('div')?.textContent).toContain(
      'BaseComponent'
    );
  });

  afterEach(async () => {
    await jest.clearAllMocks();
  });
});
