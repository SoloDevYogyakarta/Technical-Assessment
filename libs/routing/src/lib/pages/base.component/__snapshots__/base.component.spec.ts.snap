// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseComponent should render snapshot 1`] = `
<base-component
  __ngContext__="0"
  title={[Function String]}
>
  <div>
    BaseComponent is works!
  </div>
</base-component>
`;
